packages/e2e-test/tests/light-client-rpc.e2e.test.ts

```diff
@@ -64,6 +68,52 @@ async function waitLightClientSync(
   }
 }
 
+async function waitLightClientFetchTransaction(
```

Comment:
can be refactored with p-retry and p-timeout, the `await lightClientRpc.fetchTransaction` may await forever cause CI running for a long time

```ts
import pRetry from 'p-retry'
import pTimeout from 'p-timeout'

pRetry(
  () => pTimeout(
    () => {
      const result = await lightClientRpc.fetchTransaction(txHash);
      if (result.status === FetchFlag.Fetched) return result.data
      throw new Error('transaction not found')
    },
), { retries: 3, onFailedAttempt: () => asyncSleep(300) })

```